### PR TITLE
[LM-10433] Added Notification

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**

1. Added InApp Notification
2. Added push notifications 

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: likeminds-feed-reactnative-integration/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "license": "ISC",
   "dependencies": {
     "@likeminds.community/feed-js": "1.3.1",
-    "@shopify/flash-list": "1.6.3",
+    "@notifee/react-native": "7.8.2",
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/package.json:22]: Why did you remove Flashlist?

File: example/SampleApp/sampleApp/App.tsx [hunk 2]
```diff
@@ -35,20 +35,50 @@ const App = () => {
...
+
+  useEffect(() => {
+    return notifee.onForegroundEvent(({type, detail}) => {
+      let routes = getRoute(detail?.notification?.data?.route);
+      switch (type) {
+        case EventType.DISMISSED:
+          console.log('User dismissed notification', detail.notification);
+          break;
+        case EventType.PRESS:
+          if (!!navigationRef) {
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/App.tsx:56]: add comments on useEffects

File: example/SampleApp/sampleApp/android/app/build.gradle [hunk 3]
```diff
@@ -1,6 +1,7 @@
 apply plugin: "com.android.application"
 apply plugin: "org.jetbrains.kotlin.android"
 apply plugin: "com.facebook.react"
+apply plugin: "com.google.gms.google-services"
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/android/app/build.gradle:4]: We should add in the steps of integrating notification to add this line in `apply plugin: "com.google.gms.google-services"` and remove from here. By this way we can't make firebase notifications optional

File: example/SampleApp/sampleApp/android/build.gradle [hunk 4]
```diff
@@ -15,6 +15,7 @@ buildscript {
         classpath("com.android.tools.build:gradle")
         classpath("com.facebook.react:react-native-gradle-plugin")
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin")
+        classpath 'com.google.gms:google-services:4.3.15'
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/android/build.gradle:18]: this line should be add in the steps and remove it from here

File: example/SampleApp/sampleApp/feedScreen/feed.tsx [hunk 5]
```diff
@@ -63,22 +76,81 @@ const Feed = ({route}) => {
...
+      }
+    };
+    token();
+  }, []);
+
+  useEffect(() => {
+    if (FCMToken) {
+      pushAPI(FCMToken, accessToken);
+    }
+  }, [FCMToken]);
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/feedScreen/feed.tsx:131]: this firebase code should be inside a check if `react-native-firebase/messaging` exist then only you execute this code. That's how you make this library optional to use it or not. And move this code to a separate component and then call that component from this file. Take reference of initMyClient in chat

`import { playbackService } from "./audio";
import { ConversationState, LMChatClient } from "@likeminds.community/chat-rn";
import { Client } from "./client";
import AudioPlayer from "./optionalDependecies/AudioPlayer";

export const initMyClient = (
  apiKey: string,
  filterStateMessage: ConversationState[]
) => {
  const myClient = LMChatClient.setApiKey(apiKey)
    .setfilterStateConversation(filterStateMessage)
    .build();

  Client.setMyClient(myClient);

  AudioPlayer
    ? AudioPlayer?.default?.registerPlaybackService(() => playbackService)
    : null;

  return myClient;
};`

File: example/SampleApp/sampleApp/index.js [hunk 6]
```diff
@@ -5,7 +5,31 @@
...
+      setTimeout(() => {
+        RootNavigation.navigate(routes.route, routes.params); 
+      }, 2000);
+    }
+  }
+});
+
+messaging().setBackgroundMessageHandler(async remoteMessage => {
+  await getNotification(remoteMessage);
+  return remoteMessage;
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/index.js:31]: this should be optional too

File: example/SampleApp/sampleApp/ios/Podfile [hunk 7]
```diff
@@ -8,6 +8,10 @@ require Pod::Executable.execute_command('node', ['-p',
 platform :ios, min_ios_version_supported
 prepare_react_native_project!
 
+pod 'Firebase', :modular_headers => true
+pod 'FirebaseCoreInternal', :modular_headers => true
+pod 'GoogleUtilities', :modular_headers => true
+pod 'FirebaseCore', :modular_headers => true
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/Podfile:14]: dont push these keep these in a step to integrate notification. Create a doc on it. And move all the steps there

File: example/SampleApp/sampleApp/ios/sampleApp.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -14,6 +14,7 @@
 		13B07FC11A68108700A75B9A /* main.m in Sources */ = {isa = PBXBuildFile; fileRef = 13B07FB71A68108700A75B9A /* main.m */; };
 		7699B88040F8A987B510C191 /* libPods-sampleApp-sampleAppTests.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 19F6CBCC0A4E27FBF8BF4A61 /* libPods-sampleApp-sampleAppTests.a */; };
 		81AB9BB82411601600AC10FF /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 81AB9BB72411601600AC10FF /* LaunchScreen.storyboard */; };
+		DD5336D72BC02BDB00DB26F3 /* GoogleService-Info.plist in Resources */ = {isa = PBXBuildFile; fileRef = DD5336D62BC02BDB00DB26F3 /* GoogleService-Info.plist */; };
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/sampleApp.xcodeproj/project.pbxproj:17]: dont push this

File: example/SampleApp/sampleApp/ios/sampleApp.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -43,6 +44,8 @@
...
 		81AB9BB72411601600AC10FF /* LaunchScreen.storyboard */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = file.storyboard; name = LaunchScreen.storyboard; path = sampleApp/LaunchScreen.storyboard; sourceTree = "<group>"; };
 		89C6BE57DB24E9ADA2F236DE /* Pods-sampleApp-sampleAppTests.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-sampleApp-sampleAppTests.release.xcconfig"; path = "Target Support Files/Pods-sampleApp-sampleAppTests/Pods-sampleApp-sampleAppTests.release.xcconfig"; sourceTree = "<group>"; };
+		DD5336D52BC02A6C00DB26F3 /* sampleApp.entitlements */ = {isa = PBXFileReference; lastKnownFileType = text.plist.entitlements; name = sampleApp.entitlements; path = sampleApp/sampleApp.entitlements; sourceTree = "<group>"; };
+		DD5336D62BC02BDB00DB26F3 /* GoogleService-Info.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; path = "GoogleService-Info.plist"; sourceTree = "<group>"; };
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/sampleApp.xcodeproj/project.pbxproj:48]: remove it

File: example/SampleApp/sampleApp/ios/sampleApp/AppDelegate.mm [hunk 10]
```diff
@@ -1,5 +1,5 @@
 #import "AppDelegate.h"
-
+import FirebaseCore
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/sampleApp/AppDelegate.mm:2]: keep it in the step to integrate firebase

File: example/SampleApp/sampleApp/ios/sampleApp/Info.plist [hunk 11]
```diff
@@ -33,6 +33,11 @@
 	</dict>
 	<key>NSLocationWhenInUseUsageDescription</key>
 	<string></string>
+	<key>UIBackgroundModes</key>
+	<array>
+		<string>fetch</string>
+		<string>remote-notification</string>
+	</array>
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/sampleApp/Info.plist:40]: remove it, and keep it in the steps

File: example/SampleApp/sampleApp/package.json [hunk 12]
```diff
@@ -10,13 +10,18 @@
     "test": "jest"
   },
   "dependencies": {
+    "@likeminds.community/feed-js": "1.3.1",
     "@likeminds.community/feed-rn-core": "0.7.2",
+    "@notifee/react-native": "7.8.2",
     "@react-native-community/netinfo": "11.3.0",
+    "@react-native-firebase/app": "19.1.2",
+    "@react-native-firebase/messaging": "19.1.2",
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/package.json:18]: move it to peer deps and peer deps meta and make it optional

File: likeminds-feed-reactnative-integration/notification/index.ts [hunk 13]
```diff
@@ -0,0 +1,76 @@
+import { NAVIGATED_FROM_NOTIFICATION } from "../constants/Strings";
+import { CREATE_POST, POST_DETAIL, UNIVERSAL_FEED } from "../constants/screenNames";
+import notifee, {AndroidImportance, EventType} from '@notifee/react-native';
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/notification/index.ts:3]: instead of import use require check

File: likeminds-feed-reactnative-integration/screens/notificationsList/index.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                    ? onNotificationItemClickedProp(item)
+                    : handleActivityOnTap(item);
+                }}
+              />
+            );
+          }}
+          onEndReachedThreshold={0.3}
+          onEndReached={() => {
+            setNotificationFeedPageNumber(notificationFeedPageNumber + 1);
+          }}
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/screens/notificationsList/index.tsx:44]: write correct pagination logic, there is no loading flow in it. The one I discussed in morning

File: likeminds-feed-reactnative-integration/package.json (hunk 1 above)
Comment (prabha-likeminds) [reply likeminds-feed-reactnative-integration/package.json:22]: not using it
